components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt

```diff
@@ -437,26 +441,45 @@ open class Nimbus(
         }
     }
 
-    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun recordExposure(featureId: String) {
-        dbScope.launch {
-            recordExposureOnThisThread(featureId)
-        }
-    }
-
     // The exposure event should be recorded when the expected treatment (or no-treatment, such as
     // for a "control" branch) is applied or shown to the user.
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    @WorkerThread
+    @AnyThread
     internal fun recordExposureOnThisThread(featureId: String) = withCatchAll("recordExposure") {
-        val activeExperiments = getActiveExperiments()
-        activeExperiments.find { it.featureIds.contains(featureId) }?.also { experiment ->
-            NimbusEvents.exposure.record(NimbusEvents.ExposureExtra(
-                experiment = experiment.slug,
-                branch = experiment.branchSlug,
+        // First, we get the enrolled feature, if there is one, for this id.
+        val enrollment = nimbusClient.getEnrollmentByFeature(featureId) ?: return@withCatchAll
```

Comment:
Rewritten `recordExposureEvent` in terms of `getEnrollmentByFeature`. Tests still pass.